bootstore/src/db/models.rs

```diff
@@ -0,0 +1,76 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! DB models
+
+use diesel::deserialize::FromSql;
+use diesel::prelude::*;
+use diesel::serialize::ToSql;
+use diesel::FromSqlRow;
+
+use super::macros::array_new_type;
+use super::macros::json_new_type;
+use super::schema::*;
+use crate::trust_quorum::SerializableShareDistribution;
+
+json_new_type!(Share, SerializableShareDistribution);
+
+#[derive(Debug, Queryable, Insertable)]
+pub struct KeySharePrepare {
+    pub epoch: i32,
+    pub share: Share,
+    pub share_digest: Sha3_256Digest,
+}
+
+#[derive(Debug, Queryable, Insertable)]
+pub struct KeyShareCommit {
```

Comment:
I'm wondering if this table should just be a boolean `committed` field in `KeySharePrepares`. That would probably simplify queries a bit.

Reply:
Seems reasonable to me; otherwise we have the epoch/share_digest duplicated and queries have to look for both (I assume that's what you were implying with simplifying queries?). At that point does the table name become just `KeyShares`?

Reply:
Yeah, that's my reasoning. And yeah, I think it just becomes `KeyShares`.

Reply:
Fixed in https://github.com/oxidecomputer/omicron/pull/1656/commits/10058ed9b702f4d78501d99bf443189a29fbb1dd